core/chains/evm/client/send_only_node_test.go

```diff
@@ -0,0 +1,162 @@
+package client_test
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/rpc"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+
+	"github.com/smartcontractkit/chainlink/core/chains/evm/client/mocks"
+
+	"github.com/smartcontractkit/chainlink/core/assets"
+	evmclient "github.com/smartcontractkit/chainlink/core/chains/evm/client"
+	"github.com/smartcontractkit/chainlink/core/internal/testutils"
+	"github.com/smartcontractkit/chainlink/core/logger"
+)
+
+func TestNewSendOnlyNode(t *testing.T) {
+	t.Parallel()
+
+	urlFormat := "http://user:%s@testurl.com"
+	password := "pass"
+	url := testutils.MustParseURL(t, fmt.Sprintf(urlFormat, password))
+	redacted := fmt.Sprintf(urlFormat, "xxxxx")
+	lggr := logger.TestLogger(t)
+	name := "TestNewSendOnlyNode"
+	chainID := testutils.NewRandomEVMChainID()
+
+	node := evmclient.NewSendOnlyNode(lggr, *url, name, chainID)
+	assert.NotNil(t, node)
+
+	// Must contain name & url with redacted password
+	assert.Contains(t, node.String(), fmt.Sprintf("%s:%s", name, redacted))
+	assert.Equal(t, node.ChainID(), chainID)
+}
+
+func TestStartSendOnlyNode(t *testing.T) {
+	t.Parallel()
+
+	t.Run("Start with Random ChainID", func(t *testing.T) {
```

Comment:
You can/shall add `t.Parallel()` to each `t.Run`, so they will execute in parallel also.